src/commands/local.ts

```diff
@@ -610,6 +629,174 @@ function getBundler(classConfiguration: ClassConfiguration): BundlerInterface |
+            );
+            break;
+        }
+        case "redis": {
+            await runDockerContainer(
+                "redis",
+                projectName,
+                `docker run -p 6379:6379 --name genezio-redis-${projectName} -d redis `,
+            );
+            log.info(
+                `Your local redis database is up and running.\nYou can connect to the your redis instance using the following URL: redis://localhost:6379`,
+            );
+            break;
+        }
+        default: {
+            await stopDockerDatabase();
+            printAdaptiveLog(`Failed starting local docker ${database.type} database`, "");
+            log.error(
+                "Unsupported database type, please use one of the following types of databases if you want to have a local testing db: [`postgres`,`redis`]",
+            );
+            break;
+        }
+    }
+}
+
+async function stopDockerContainer(databaseType: string, projectName: string) {
+    const containerName = `genezio-${databaseType}-${projectName}`;
+    let output;
+    try {
+        output = await asyncExec(`docker ps -a --format "{{.Names}}" `);
+    } catch (error: any) {
+        log.error(`An error has occured: ${error.toString()}`);
+        log.error(
+            "This is most likely a docker error, if you want to solve it you can check out their documentation at https://docs.docker.com/ ",
+        );
+        return undefined;
+    }
+    if (output) {
+        const containerNames = output.stdout.trim().split(`\n`);
+        if (containerNames.includes(containerName)) {
+            let isContainerRunningOutput;
+            try {
+                isContainerRunningOutput =
+                    await asyncExec(`docker inspect -f '{{.State.Running}}' ${containerName}
+                `);
+            } catch (error: any) {
+                log.error(`An error has occured: ${error.toString()}`);
+                return undefined;
+            }
+            const isContainerRunning =
+                isContainerRunningOutput.stdout.trim().replaceAll("'", "") === "true";
+            if (isContainerRunning) {
+                try {
+                    await asyncExec(`docker stop ${containerName} `);
+                } catch (error: any) {
+                    log.error(`An error has occured: ${error.toString()}`);
+                    return undefined;
+                }
+            }
+        }
+    }
+}
+
+export async function stopDockerDatabase() {
+    const yamlProjectConfiguration = await getProjectConfiguration();
+    if (yamlProjectConfiguration.database?.type) {
+        try {
+            await asyncExec("docker --version");
+        } catch (error: any) {
+            return undefined;
+        }
+        switch (yamlProjectConfiguration.database?.type) {
```

Comment:
i feel like this is duplicated and not needed

Reply:
I think what @radu1122 is saying is that all the cases are doing the same thing. What was the intention?